plugins/filter_aws/aws.c

```diff
@@ -370,6 +400,208 @@ static int get_vpc_metadata(struct flb_filter_aws *ctx)
     return ret;
 }
 
+void flb_filter_aws_tags_destroy(struct flb_filter_aws *ctx)
+{
+    size_t i;
+    if (!ctx) {
+        return;
+    }
+    if (ctx->tag_keys) {
+        for (i = 0; i < ctx->tags_count; i++) {
+            if (ctx->tag_keys[i]) {
+                flb_sds_destroy(ctx->tag_keys[i]);
+            }
+        }
+        flb_free(ctx->tag_keys);
+        ctx->tag_keys = NULL;
+    }
+    if (ctx->tag_values) {
+        for (i = 0; i < ctx->tags_count; i++) {
+            if (ctx->tag_values[i]) {
+                flb_sds_destroy(ctx->tag_values[i]);
+            }
+        }
+        flb_free(ctx->tag_values);
+        ctx->tag_values = NULL;
+    }
+    if (ctx->tag_keys_len) {
+        flb_free(ctx->tag_keys_len);
+    }
+    ctx->tag_keys_len = NULL;
+    if (ctx->tag_values_len) {
+        flb_free(ctx->tag_values_len);
+    }
+    ctx->tag_values_len = NULL;
```

Comment:
same comment here. Size_t is just a long unsigned int of some sort, its zero value is `0`.

Reply:
Again, `size_t *tag_values_len` is a pointer, it's "zero" value should be `NULL`. We should set this to `NULL` to point out that there is no memory initialized.

Reply:
Yea sorry I got confused somehow. You're right.